Sources/Dependencies/FilterDependencyContainer.swift

```diff
@@ -5,6 +5,8 @@
 import Foundation
 
 public class FilterDependencyContainer {
+    public weak var applyButtonDelegate: ApplySelectionButtonDelegate?
```

Comment:
Maybe it's better to pass it as a parameter in `makeMultiLevelListSelectionFilterViewController` since it's the only place it's used.

Reply:
You are right, you could even set the delegate in the `navigate(to destination:)` function instead of passing it.